### ISSUE TITLE
Allow users to set port description logic per peer_type in eos_designs

### ISSUE THREAD

**Issue by ryanmerolle** [aristanetworks/ansible-avd#995]
**Issue Type**

- [X] Template enhancement
- [X] Role enhancement
- [X] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

You cannot custom_structured_configuration to only update the interface descriptions on ethernet_interfaces or port_channel_interfaces.  The only way you could do this is loop through each interface, copy said interface values to a new dictionary and rewrite the description key to use new logic.

**Describe the solution you'd like**

Being able to define interface description logic per peer_type would be huge.  It should allow strings and key names in creating the description logic.  This could be further extended in the future if for example someone used NetBox to pull in circuit information for WAN/DCI links to add carrier and circuit info to the associated interface dictionary.  Assuming circuit info is present, the interface description logic could use that to build better descriptions.

**Describe alternatives you've considered**

Use custom_structured_configuration to replace the full interface dictionaries as a hack.

**Additional context**

@carlbuchmann and I were talking about this, and he suggested an issue be raised.

**Comment by ClausHolbechArista** [comment]
We have a new feature incoming for `structured_config` under various objects in `eos_designs`. So you will be able to modify/override the generated `structured_configuration` to fit your own logic. Take a look at PR #900. I expect that it will be part of AVD 3.0, and we will merge it to `devel` soon after 2.2 has been released.

**Comment by ryanmerolle** [comment]
Hard to follow PR #900 as there is no explanation of the feature(s) being proposed, linking issue explaining the approach, or doc updates.  I get its a draft, but before mentioning it in response to this issue it should probably get some more context associated to it first.  Totally not a big deal, just saying how other contributors/users could be confused.

It feels like you are adding some better control of dictionary/functional config specific logic beyond the current implementation of custom_sctructured_config.

**Comment by ClausHolbechArista** [comment]
Valid point. I will update the PR description on #900 to reflect the purpose and data-model.

**Comment by ClausHolbechArista** [comment]
A simple solution to this, would be to expose a `description` keys on the adapters, like we have on the port-channel, to allow the user to set the description themselves. If not set, we will use the generated one.
